govuk_chat_evaluation/rag_answers/custom_deepeval/metrics/coherence/coherence.py

```diff
@@ -0,0 +1,131 @@
+        verbose_mode: bool = False,
+    ):
+        self.threshold = 1.0 if strict_mode else threshold
+        self.include_reason = include_reason
+        self.strict_mode = strict_mode
+        self.verbose_mode = verbose_mode
+
+        self.model, self.using_native_model = initialize_model(model)
+        self.evaluation_model = self.model.get_model_name()
+        self.rubric_score: int | None = None
+
+    async def a_measure(
+        self,
+        test_case: LLMTestCase,
+        _show_indicator: bool = True,
+        _in_component: bool = False,
+    ) -> float:
+        check_llm_test_case_params(test_case, self._required_params, self)
+
+        if self.using_native_model:
+            self.evaluation_cost = 0.0
+
+        with metric_progress_indicator(
+            self,
+            async_mode=True,
+            _show_indicator=_show_indicator,
+            _in_component=_in_component,
+        ):
+            self.error = None
+            prompt = self.evaluation_template.evaluate(
+                user_input=test_case.input or "",
+                actual_output=test_case.actual_output or "",
+            )
+
+            judgement = await self._generate_result_from_model(
+                prompt, schema=CoherenceJudgement
+            )
+
+            self.rubric_score = self._clamp_score(judgement.score)
+            self.score = self._normalize(self.rubric_score)
+            cleaned_reason = judgement.reason.strip()
+            self.reason = cleaned_reason if self.include_reason else None
+            self.success = self.is_successful()
+
+            capture_metric_type(
+                self.__name__, async_mode=True, in_component=_in_component
+            )
+
+            verbose_reason = cleaned_reason if self.include_reason else "Reason omitted"
+            self.verbose_logs = construct_verbose_logs(
+                self,
+                steps=[
+                    f"Rubric Score: {self.rubric_score}",
+                    f"Score: {self.score}",
+                    f"Reason: {verbose_reason}",
+                ],
+            )
+
+            return self.score
+
+    async def _generate_result_from_model(
+        self, prompt: str, schema: Type[CoherenceJudgement]
+    ) -> CoherenceJudgement:
+        if self.using_native_model:
+            result, cost = cast(
+                tuple[CoherenceJudgement, float],
+                await self.model.a_generate(prompt, schema=schema),
+            )
+            self.evaluation_cost = (self.evaluation_cost or 0.0) + cost
+            return result
+
+        try:
+            return cast(
+                CoherenceJudgement,
+                await self.model.a_generate(prompt, schema=schema),
+            )
+        except TypeError:
+            raw_response = await self.model.a_generate(prompt)
+            data = trimAndLoadJson(raw_response, self)
+            return schema(**data)
+
+    def _normalize(self, raw_score: int) -> float:
```

Comment:
```suggestion
    def _normalize_score(self, rubric_score: int) -> float:
```

we still have a raw_score here and I think this method would better signify intent with a _score suffix of what it is normalising